File: packages/ui-compose/ui/src/test/kotlin/com/kinetixui/ui/blocks/StatCardsBlockTest.kt
package com.kinetixui.ui.blocks

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.TrendingDown
import androidx.compose.material.icons.filled.TrendingUp
import androidx.compose.material3.Icon
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.test.junit4.v2.createComposeRule
import androidx.compose.ui.test.onNodeWithText
import androidx.compose.ui.unit.dp
import com.kinetixui.ui.KinetixMetric
import com.kinetixui.ui.KinetixMetricTrend
import com.kinetixui.ui.KinetixTheme
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

// The "Stat cards" block — see TestimonialBlockTest.kt for how block fixtures work.
//
// kx-block:start
@Composable
fun StatCardsBlock() {
    Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
        KinetixMetric(
            label = "Revenue",
            value = "$45,231",
            modifier = Modifier.weight(1f),
            trend = KinetixMetricTrend.Up,
            change = "12.5%",
            icon = { Icon(Icons.Default.TrendingUp, contentDescription = null) },
        )
        KinetixMetric(
            label = "Active users",
            value = "2,420",
            modifier = Modifier.weight(1f),
            trend = KinetixMetricTrend.Up,
            change = "8.1%",
            icon = { Icon(Icons.Default.TrendingUp, contentDescription = null) },
        )
        KinetixMetric(
            label = "Churn",
            value = "1.2%",
            modifier = Modifier.weight(1f),
            trend = KinetixMetricTrend.Down,
            change = "0.3%",
            icon = { Icon(Icons.Default.TrendingDown, contentDescription = null) },
        )
    }
}
// kx-block:end

@RunWith(RobolectricTestRunner::class)
@Config(sdk = [34])
class StatCardsBlockTest {
    @get:Rule
    val rule = createComposeRule()

    @Test
    fun renders_every_metric_with_its_value_and_change() {
        rule.setContent { KinetixTheme(darkTheme = false) { StatCardsBlock() } }
        rule.onNodeWithText("Revenue").assertExists()
        rule.onNodeWithText("2,420").assertExists()
        rule.onNodeWithText("Churn").assertExists()
        rule.onNodeWithText("0.3%").assertExists()
    }
}

File: packages/ui-flutter/test/blocks/stat_cards_block_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:kinetix_ui/kinetix_ui.dart';

// The "Stat cards" block — see testimonial_block_test.dart for how block fixtures work.
//
// kx-block:start
class StatCardsBlock extends StatelessWidget {
  const StatCardsBlock({super.key});

  @override
  Widget build(BuildContext context) {
    return const Row(
      children: <Widget>[
        Expanded(
          child: KinetixMetric(
            label: 'Revenue',
            value: r'$45,231',
            trend: KinetixMetricTrend.up,
            change: '12.5%',
            icon: Icon(Icons.trending_up, size: 16),
          ),
        ),
        SizedBox(width: 16),
        Expanded(
          child: KinetixMetric(
            label: 'Active users',
            value: '2,420',
            trend: KinetixMetricTrend.up,
            change: '8.1%',
            icon: Icon(Icons.group_outlined, size: 16),
          ),
        ),
        SizedBox(width: 16),
        Expanded(
          child: KinetixMetric(
            label: 'Churn',
            value: '1.2%',
            trend: KinetixMetricTrend.down,
            change: '0.3%',
            icon: Icon(Icons.trending_down, size: 16),
          ),
        ),
      ],
    );
  }
}
// kx-block:end

void main() {
  testWidgets('renders every metric with its value and change', (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: KinetixTheme(
          brightness: Brightness.light,
          child: const Scaffold(body: StatCardsBlock()),
        ),
      ),
    );

    expect(find.text('Revenue'), findsOneWidget);
    expect(find.text('2,420'), findsOneWidget);
    expect(find.text('Churn'), findsOneWidget);
    expect(find.text('0.3%'), findsOneWidget);
  });
}
